sonar-iac-plugin/src/test/java/org/sonar/iac/ArchUnitTest.java

```diff
@@ -61,4 +65,37 @@ public void check(JavaClass javaClass, ConditionEvents events) {
    */
   @ArchTest
   static final ArchRule shouldNotCallPathToRealPath = noClasses().should().callMethod(Path.class, "toRealPath", LinkOption[].class);
+
+  @ArchTest
+  static final ArchRule shouldOverrideChildrenMethodIfContainsNodes = classes().that().implement(Node.class)
```

Comment:
Nice! I also thought about adding an arch unit test to enforce `fromPb` method on all nodes, but didn't do it